### PR TITLE
Use mariadb UUID when specify UUID type

### PR DESCRIPTION
**Pull request by mohamed-ahrrass-nw**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description 
fix : https://github.com/liquibase/liquibase/issues/5263
Add UUID mariadb type when using UUID as a field type

### CODE REVIEW

**Comment by filipelautert** [comment]
@mohamed-ahrrass-nw  would this work for Mysql too?

**Comment by MalloD12** [comment]
Hi @mohamed-ahrrass-nw,

Thanks for submitting this PR. Your code changes look good to me. I have added a few more changes to yours that you can see in my last commit. These are:

- Added a version check so we don't break deployment for people using a version lower than 10.7. 
- Test updates.

Thanks,
Daniel.

**Comment by mohamed-ahrrass-nw** [comment]
Hi @MalloD12 ,

Thank you for your return. I will investigate the failing tests to understand their connection to the changes made. If it's something on our end, I'll address it promptly. 

Please let me know if there's any specific information or updates about the subject.

Thanks 
Mohamed

**Comment by MalloD12** [comment]
> Details

Hi @mohamed-ahrrass-nw,

It's not something related to these changes. I found our team has created this [ticket](https://datical.atlassian.net/browse/DAT-16429) (Internal), and I hope we can address it soon. Once that's done and we have a green check in all our test jobs I'll approve this PR and pass it to other team members, so they can review it and do a more in-depth testing.

Thanks,
Daniel.

**Comment by mohamed-ahrrass-nw** [comment]
Hi @MalloD12 
Okay, Thank you Daniel

Mohamed

**Review by MalloD12 — APPROVED** [review]
**Approved**.

## Review and testing results:
Code changes looked good to me. Anyway, a minor changed has been introduced to make sure UUID type is returned for MariaDB version equals or greater than `10.7`. Also, MariaDB version has been updated for testing, and finally added a unit test to validate this change (uuid is returned for a MariaDB).

Build and test checks have been successfully executed.

**Comment by mohamed-ahrrass-nw** [comment]
Hello @MalloD12 ,
Thank you very much for your return. 👍 
Mohamed

**Comment by MalloD12** [comment]
> Hello @MalloD12 , Thank you very much for your return. 👍 Mohamed

Thank you, @mohamed-ahrrass-nw for your contribution!

Daniel.

File: liquibase-standard/src/main/java/liquibase/datatype/core/UUIDType.java [hunk 1]
```diff
@@ -36,6 +36,18 @@ public DatabaseDataType toDatabaseDataType(Database database) {
         if (database instanceof SQLiteDatabase) {
             return new DatabaseDataType("TEXT");
         }
+        try {
+            if(database instanceof MariaDBDatabase && (database.getDatabaseMajorVersion() == 10 && database.getDatabaseMinorVersion() >= 7)
+                    || database.getDatabaseMajorVersion() > 11) {
```
[the remark this answers is not in the document]
Comment (MalloD12) [reply liquibase-standard/src/main/java/liquibase/datatype/core/UUIDType.java:41]: Agree, it should be `>=`. Thank you!

[the remark this answers is not in the document]
Comment (MalloD12) [reply liquibase-standard/src/main/java/liquibase/datatype/core/UUIDType.java]: Done

**Review by filipelautert — APPROVED** [review]

**Comment by mohamed-ahrrass-nw** [comment]
Hello @MalloD12 ,
I see that CodeQL is failing because of an error related to token : `Error: Input required and not supplied: token`, is it something that we can fix in our end ? Or the opposite ? 
Thank you 
Mohamed !

**Comment by filipelautert** [comment]
Hi @mohamed-ahrrass-nw - we can ignore codeql for now, no worries. 
Once QA @rberezen is back from xmas break he can review this pr and then we can get it merged.

**Comment by mohamed-ahrrass-nw** [comment]
> Hi @mohamed-ahrrass-nw - we can ignore codeql for now, no worries. Once QA @rberezen is back from xmas break he can review this pr and then we can get it merged.

Hello @filipelautert,
Okay, thank you for your return 
Thanks 
Mohamed

**Review by rberezen — APPROVED** [review]

**Comment by adrian-velonis1** [comment]
Internal doc ticket: https://datical.atlassian.net/browse/PD-4393

**Pull request merged**